### PR TITLE
🏠  Home page design

### PR DESCRIPTION
**Pull request by Pinimes**
## Status 
WIP

## Description
Enhanced the home page's user interface for Theology 101 website and incorporated a showcase picture.

## Screenshot

homepage

### CODE REVIEW

File: styles/common.module.scss [hunk 1]
```diff
@@ -39,14 +40,60 @@
...
+    background-size: cover;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    color: white;
+    font-weight: 900;
+
+    h1 {
+        font-size: 80px;
+        font-family: cursive;
```
Comment (jontsai) [inline styles/common.module.scss:79]: Not sure why this is showing Comic Sans. Let's pick someone from https://fonts.google.com/

**Review by jontsai — CHANGES_REQUESTED** [review]

**Review by jontsai — APPROVED** [review]
LGTM

**Review by YosefAshenafi — APPROVED** [review]
LGTM

**Review by lleung030 — APPROVED** [review]
LGTM

File: pages/_app.jsx [hunk 2]
```diff
@@ -1,18 +1,26 @@
...
-
+import { useRouter } from 'next/router';
 import { BaseLayout } from '@components/base_layout';
 import 'bootstrap/dist/css/bootstrap.min.css';
 import '@styles/globals.scss';
+import Home from '.';
 
 function App({ Component, pageProps }) {
+    const router = useRouter();
+    const isHomePage = router.pathname === '/';
```
Comment (goztrk) [inline pages/_app.jsx:10]: This is being repeated in two places so it is not DRY.

To make it DRY, we need to create a custom hook (it is actually a function this time but hooks are a special trait of React)

- Create a file named `utils/hooks.js`
- Put the following code inside it:

```js
import { useRouter } from 'next/router';

function useIsHomePage() {
    const router = useRouter();
    return router.pathname === '/';
}
```

Then you can use this function/hook here:
```suggestion
    const isHomePage = useIsHomePage();
```

Don't forget to add `import { useIsHomePage() } from '@utils/hooks';`.
Also, Remove the unused `useRouter` import.

File: pages/_app.jsx [hunk 3]
```diff
@@ -1,18 +1,26 @@
...
         // Correct way to import bootstrap JS in the client side
         require('bootstrap/dist/js/bootstrap.bundle.min.js');
     }, []);
 
-    return <BaseLayout>{getLayout(<Component {...pageProps} />)}</BaseLayout>;
+    return isHomePage ? (
+        <Home />
+    ) : (
+        <BaseLayout>{getLayout(<Component {...pageProps} />)}</BaseLayout>
+    );
```
Comment (goztrk) [inline pages/_app.jsx:23]: This is not correct usage. If you do not use `BaseLayout` component, the header definitions on `BaseLayout` (meta, favicon, font-awesome, google, analytics) will not be applied to the homepage.

First let's change this to:

```suggestion
    return isHomePage ? (
        <BaseLayout hideSidebar={isHomePage}>{getLayout(<Component {...pageProps} />)}</BaseLayout>
    );
```

Then in `components/base_layout.jsx`:

```jsx
...
export function BaseLayout({ hideSidebar }) {
...
                           <div className={hideSidebar ? "col-12" : "col-lg-9 col-md-8"}>{children}</div>
{!hideSidebar && (
                           <div className="col-lg-3 col-md-4">
                                <div className="bg-light text-dark w-100 p-3">
                                    <SidebarList
                                        title="Topics"
                                        url="topics"
                                        entries={topics}
                                    />
                                    <SidebarList
                                        title="Tags"
                                        url="tags"
                                        entries={tags}
                                    />
                                </div>
                                <a
                                    href="https://theology101ak.thinkific.com/courses/howtostudythebible?ref=6431a0"
                                    target="_blank"
                                >
                                    <img className="img-fluid" src="/ad.png" />
                                </a>
                            </div>
)}
```
With this, we made the sidebar dynamic. We can disable showing it with a prop depending on the page.

File: pages/index.jsx
Comment (goztrk) [inline pages/index.jsx:1]: Revert the head changes on this page, please. See the comment above.

**Review by goztrk — CHANGES_REQUESTED** [review]

**Review by goztrk — APPROVED** [review]
LGTM

**Pull request merged**